### PR TITLE
[521918] Handle SWTError

### PR DESCRIPTION
**Pull request by kthoms**
Signed-off-by: Karsten Thoms <karsten.thoms@itemis.de>

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=521918

File: org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/editor/actions/ImportsAwareClipboardAction.java [hunk 1]
```diff
@@ -140,31 +141,35 @@ private void doCutCopyWithImportsOperation() {
...
+								payload.put(rtfData, rtfTransfer);
+							}
+							
+							List<Object> datas = newArrayList();
+							List<Transfer> dataTypes = newArrayList();
+							for (Entry<Object, Transfer> entry : payload.entrySet()) {
+								datas.add(entry.getKey());
+								dataTypes.add(entry.getValue());
+							}
+							clipboard.setContents(datas.toArray(), dataTypes.toArray(new Transfer[] {}));
```
Comment (cdietrich) [inline org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/editor/actions/ImportsAwareClipboardAction.java]: should we have a look at the error and if the e.code == DND.ERROR_CANNOT_SET_CLIPBOARD show a dialog?

Comment (kthoms) [reply org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/editor/actions/ImportsAwareClipboardAction.java]: And what to present? Further Clipboard is platform specific.
I don't think that this is worth handling in more detail.

Comment (cdietrich) [reply org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/editor/actions/ImportsAwareClipboardAction.java]: have a look at org.eclipse.debug.internal.ui.viewers.model.VirtualCopyToClipboardActionDelegate.writeBufferToClipboard(StringBuffer) or org.eclipse.jdt.internal.ui.actions.CopyQualifiedNameAction.run()

Comment (kthoms) [reply org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/editor/actions/ImportsAwareClipboardAction.java]: Good point. I was not ware that other actions do this also. I will adjust this accordingly.

Comment (kthoms) [reply org.eclipse.xtext.xbase.ui/src/org/eclipse/xtext/xbase/ui/editor/actions/ImportsAwareClipboardAction.java]: Updated.

**Pull request merged**